From ad76955a431baca2a8b6cc875b2633db9f27b12e Mon Sep 17 00:00:00 2001
From: Jonathan Lebon <jonathan@jlebon.com>
Date: Fri, 28 Aug 2020 17:27:24 -0400
Subject: [PATCH] fetch: don't run if fetch-offline fetched a config

It's possible that `fetch-offline` does successfully fetch and cache a
config but still touch `/run/ignition/neednet`. That stamp file is an
API between Ignition and the OS that Ignition will need networking to
fully process it. But the config itself might've successfully been
fetched (and the only thing that requires networking is e.g. a remote
file). In those cases, there's no point in running `fetch` and have it
query the provider again.
---
 dracut/30ignition/ignition-fetch.service | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/dracut/30ignition/ignition-fetch.service b/dracut/30ignition/ignition-fetch.service
index 815208b4..c4b3e363 100644
--- a/dracut/30ignition/ignition-fetch.service
+++ b/dracut/30ignition/ignition-fetch.service
@@ -6,6 +6,8 @@ DefaultDependencies=false
 Before=ignition-complete.target
 After=basic.target
 ConditionPathExists=/run/ignition/neednet
+# Don't run if the `fetch-offline` stage successfully fetched a config
+ConditionPathExists=!/run/ignition.json
 
 # Stage order: setup -> fetch-offline [-> fetch] -> disks -> mount -> files.
 # We run after the setup stage has run because it may copy in new/different
-- 
2.21.1

